### PR TITLE
[feat-5427]: attachment viewer caption

### PR DESCRIPTION
**Pull request by vdegraaf**
Context
The AttachmetViewer is changed in order to show the caption and 'openbaar getoond' in the back-office or 'foto melder' in mijn-meldingen. Styling is a little bit different then from design, but discussed with Linda.

Changes

- if civilian has uploaded image, then show 'melder' in back office, otherwise show 'openbaar getoond' if image is public and if that's also not applicable, show the e-mailaddress of the civil servant who has uploaded the image.
- in mijn-meldingen always show 'foto gemeente'.
- caption is as wide as the image and as high as needed (not limited to 72 px as in the design)

Ticket: [SIG-5427](https://gemeente-amsterdam.atlassian.net/browse/SIG-5427)
Signalen

Before opening a pull request, please ensure:

    Make sure your PR title follows naming conventions: [feat-1234]: name feature
    Double-check your branch is based on main and targets main
    Pull request has tests (we are going for 100% coverage!)
    Code is well-commented, linted and follows project conventions
    Committed source code is headed by the correct SPDX license expression

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

### CODE REVIEW

File: src/components/AttachmentViewer/styles.ts [hunk 1]
```diff
@@ -129,7 +112,7 @@ export const NextButton = styled(StyledButton)`
 `
 
 export const StyledInteractiveImage = styled(InteractiveImage)`
-  margin: ${themeSpacing(23, 0, 5)};
-  max-height: calc(100vh - ${themeSpacing(28)});
+  margin: ${themeSpacing(6, 0, 0)};
+  max-height: calc(100vh - ${themeSpacing(75)});
```
Comment (vdegraaf) [inline src/components/AttachmentViewer/styles.ts:116]: height in pixels

**Review by vdegraaf — COMMENTED** [review]
Looks good, some comments. Mainly on how it turns out on mijn-meldingen:

- I think text is larger or H5 in the design instead of 16px
- Something's going wrong in mijn-meldingen. Looks like all attachments are shown double and out of proportion. Also, they are mixed together. See attachments.
- Also, I can click and view (some of) the attachments of the melder on real size, but not the attachments of de behandelaar. 
- I would expect a pointer when an image is clickable

This PR:
Screenshot 2023-09-14 at 14 47 48

ACC:
Screenshot 2023-09-14 at 14 48 39

File: src/signals/my-incidents/components/IncidentsDetail/IncidentsDetail.test.tsx [hunk 2]
```diff
@@ -66,22 +67,32 @@ describe('IncidentsDetail', () => {
     )
 
     expect(screen.getByText(/Foto's/)).toBeInTheDocument()
+  })
+
+  it('should hide img and gebeurt het vaker if props are missing', function () {
```
[the remark this answers is not in the document]
Comment (SireeKoolenWijkstra) [reply src/signals/my-incidents/components/IncidentsDetail/IncidentsDetail.test.tsx:72]: "gebeurt het vaker" is an extraProperty label.

File: src/components/AttachmentViewer/styles.ts (hunk 1 above)
Comment (SireeKoolenWijkstra) [reply src/components/AttachmentViewer/styles.ts:116]: This one I don't get. Everything is with themespacing?

Comment (SireeKoolenWijkstra) [reply src/components/AttachmentViewer/styles.ts:116]: changed it.

File: src/signals/my-incidents/components/IncidentsDetail/IncidentsDetail.tsx [hunk 3]
```diff
@@ -30,6 +33,20 @@ export const IncidentsDetail = ({
 }: Props) => {
   const { _display, text, location, extra_properties, _links } = incidentsDetail
   const attachments = _links?.['sia:attachments']
+  const [selectedAttachment, setSelectedAttachment] = useState<string | null>(
+    null
+  )
+  const [selectedAttachmentEntity, setSelectedAttachmentEntity] =
+    useState<typeof attachments>()
```
Comment (vdegraaf) [inline src/signals/my-incidents/components/IncidentsDetail/IncidentsDetail.tsx:40]: There exist a `Attachment` type in /types

**Review by vdegraaf — COMMENTED** [review]
Looking good! 

Just this padding looks too big to me, but take it up with Linda. 
Screenshot 2023-09-18 at 11 41 43

File: src/components/AttachmentViewer/AttachmentViewer.test.tsx [hunk 4]
```diff
@@ -300,4 +302,14 @@ describe('<AttachmentViewer />', () => {
       expect(onClose).toHaveBeenCalledTimes(1)
     })
   })
+
+  it('should show caption', async () => {
```
[the remark this answers is not in the document]
Comment (SireeKoolenWijkstra) [reply src/components/AttachmentViewer/AttachmentViewer.test.tsx:306]: removed it

File: src/components/AttachmentViewer/components/InteractiveImage/styled.ts [hunk 5]
```diff
@@ -36,3 +38,18 @@ export const ZoomedImage = styled.div.attrs<ZoomedImageProps>(
...
+  display: inline-block;
+  margin: unset;
+`
+
+export const StyledFigCaption = styled.figcaption`
+  background-color: ${themeColor('tint', 'level1')};
+  height: 100%;
+  font-weight: 700;
+  width: 0;
+  min-width: calc(100% - 40px);
```
[the remark this answers is not in the document]
Comment (SireeKoolenWijkstra) [reply src/components/AttachmentViewer/components/InteractiveImage/styled.ts:51]: yup, made it into a constant.

File: src/signals/my-incidents/components/IncidentsDetail/IncidentsDetail.tsx (hunk 3 above)
Comment (SireeKoolenWijkstra) [reply src/signals/my-incidents/components/IncidentsDetail/IncidentsDetail.tsx:40]: Yup, but there is a story in this sprint that will take care of all the different places that we define Attachment. It happens at multiple places that we redefine Attachment again. Changing it here is out-of-scope.

**Review by vdegraaf — COMMENTED** [review]
NICE! approved

**Pull request merged**